### PR TITLE
Validators use TLS

### PR DESCRIPTION
**Pull request by christos-h**
## Motivation

We would like to deploy validators with Ingresses on GCP with TLS enabled. Furthermore we would like TLS to be configurable depending on if the network is defined inside a validator or between a validator proxy and a client.

Finally we would like for GCP Ingresses to own certificates against which TLS connections can be established.

## Proposal

Use GCP Ingresses along with TLS on the client and proxy (with a self-signed cert). Certificates on the ingress are managed by GCP and are provisioned automatically when a validator is provisioned. GCP works with a CA to sign the certificates which takes 10-30 minutes (although your mileage my vary).

TLS on the proxy is required for GCP's HTTP/2 LB. Although the LB does not check if the proxy certificate is signed by any known CA, so a self-signed certificate which is generated in-process will suffice.

Furthermore, configuration files support specifying TLS or cleartext when using GRPC. Cleartext is preferred for local testing, and also networking inside the validator, for example between the proxy and shards.

Finally - the devnet validator configuration (previously under `configuration/prod` has been moved as it is Linera specific. It now resides inside a private repository and is no longer public.

## Test Plan

The `kubernetes` workflow acts as a regression test for the non-TLS changes. Automated testing against GCP needs some design and will be implemented in a subsequent PR.

It has been tested manually though.

### CODE REVIEW

File: kubernetes/linera-validator/templates/prometheus.yaml [hunk 1]
```diff
@@ -22,16 +22,16 @@ kind: ServiceMonitor
 metadata:
   labels:
     release: linera-core
-  name: validator-1
+  name: proxy
```
Comment (ma2bd) [inline kubernetes/linera-validator/templates/prometheus.yaml:25]: Curious what's happening here

File: kubernetes/linera-validator/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+- `Chart.yaml`: The validator's helm manifest. Defines versions, dependencies,
+  etc.
+- `Chart.lock`: A lock file for the chart's dependencies.
+- `charts/`:  Holds the source for the validator chart's dependencies.
+- `templates/`: The Kubernetes manifests templates which Helm uses to
+  parameterise the validators.
+- `working`: A directory which is in `.gitignore` (but not by Helm, i.e. it is
+  *not* in the .helmignore) used to store intermediate artifacts for
+  bootstrapping networks. For example gensis configuration, server
```
Comment (ma2bd) [inline kubernetes/linera-validator/README.md:19]: typo: gensis

**Review by ma2bd — APPROVED** [review]

File: kubernetes/linera-validator/templates/prometheus.yaml (hunk 1 above)
Comment (christos-h) [reply kubernetes/linera-validator/templates/prometheus.yaml:25]: The proxy service in Kubernetes was originally named 'validator-1' as an oversight. It has subsequently been renamed to proxy.

File: linera-service/src/server.rs [hunk 3]
```diff
@@ -199,7 +199,8 @@ impl Runnable for ServerContext {
             NetworkProtocol::Simple(protocol) => {
                 self.spawn_simple(&listen_address, states, protocol).await?
             }
-            NetworkProtocol::Grpc => self.spawn_grpc(&listen_address, states).await?,
+            // We don't support TLS between the proxy and shards.
+            NetworkProtocol::Grpc { .. } => self.spawn_grpc(&listen_address, states).await?,
```
[the remark this answers is not in the document]
Comment (christos-h) [reply linera-service/src/server.rs:203]: Good idea.

**Pull request merged**